pkg/codegen/testing/test/testdata/external-resource-schema/python/pulumi_example/_inputs.py

```diff
@@ -37,9 +37,6 @@ def __init__(__self__, *,
     @staticmethod
     def _configure(
              _setter: Callable[[Any, Any], None],
-             required_name: pulumi.Input['pulumi_random.RandomPet'],
-             required_name_array: pulumi.Input[Sequence[pulumi.Input['pulumi_random.RandomPet']]],
-             required_name_map: pulumi.Input[Mapping[str, pulumi.Input['pulumi_random.RandomPet']]],
              age: Optional[pulumi.Input[int]] = None,
              name: Optional[pulumi.Input['pulumi_random.RandomPet']] = None,
              name_array: Optional[pulumi.Input[Sequence[pulumi.Input['pulumi_random.RandomPet']]]] = None,
```

Comment:
It's strange to have the optional args as args and not have the required args (relying instead on getting the required args from kwargs).

Could we have all args listed, but make the required args emitted as optional with a default value of None, the same as the optional args?

```python
    @staticmethod
    def _configure(
             _setter: Callable[[Any, Any], None],
             required_name: Optional[pulumi.Input['pulumi_random.RandomPet']] = None, # <-- Optional in the arg list
             required_name_array: Optional[pulumi.Input[Sequence[pulumi.Input['pulumi_random.RandomPet']]]] = None,  # <-- Optional in the arg list
             required_name_map: Optional[pulumi.Input[Mapping[str, pulumi.Input['pulumi_random.RandomPet']]] = None, # <-- Optional in the arg list
             age: Optional[pulumi.Input[int]] = None,
             name: Optional[pulumi.Input['pulumi_random.RandomPet']] = None,
             name_array: Optional[pulumi.Input[Sequence[pulumi.Input['pulumi_random.RandomPet']]]] = None,
             name_map: Optional[pulumi.Input[Mapping[str, pulumi.Input['pulumi_random.RandomPet']]]] = None,
             opts: Optional[pulumi.ResourceOptions] = None,
             **kwargs):
        if required_name is None and 'requiredName' in kwargs:
            required_name = kwargs['requiredName']
        if required_name is None:
            raise TypeError("Missing 'required_name' argument")

        # ...
```

Other suggested changes from the above snippet:

1: Only look for `requiredName` if `required_name` isn't specified:

```python
        if required_name is None and 'requiredName' in kwargs:
            required_name = kwargs['requiredName']
```

2: Nit: Minor tweak to the message:

```python
 raise TypeError("Missing 'required_name' argument")
```

Reply:
I don't believe we need the Optional on required arguments.